### PR TITLE
Add useful docstring for Poly class

### LINKED ISSUE #11427: The Poly class needs a useful doc string
At the terminal I get:

```
In [1]: Poly?
Init signature: Poly(rep, *gens, **args)
Docstring:      Generic class for representing polynomial expressions. 
File:           ~/miniconda3/lib/python3.5/site-packages/sympy/polys/polytools.py
Type:           ManagedProperties
```

which is pretty useless for determining how to initialize and use the class.

### PR DESCRIPTION
**Pull request by jtoka**
Fixes #11427

Add useful docstring for Poly class.

### CODE REVIEW

**Comment by jtoka** [comment]
@moorepants please have a look when you have a chance.

File: sympy/polys/polytools.py [hunk 1]
```diff
@@ -61,7 +61,24 @@
...
+    """Generic class for representing and operating on polynomial expressions.
+
+        Examples
+        ========
+        >>> from sympy import Poly
+        >>> from sympy.abc import x, y
+        >>> Poly(x*(x**2 + x - 1)**2) # Univariate polynomial
+        Poly(x**5 + 2*x**4 - x**3 - 2*x**2 + x, x, domain='ZZ')
+
+        >>> Poly(x*(x**2 + x - 1)**2, domain = 'R') # Univariate polynomial with specified domain
```
Comment (smichr) [inline sympy/polys/polytools.py]: Go ahead an put this as text between the above and below `>>>` lines. You can feel free to say more, too, if you want. (You can see many examples of this throughout SymPy.)

**Comment by jtoka** [comment]
@smichr Updated as requested.

File: sympy/polys/polytools.py [hunk 2]
```diff
@@ -61,7 +61,40 @@
 
 @public
 class Poly(Expr):
-    """Generic class for representing polynomial expressions. """
```
Comment (smichr) [inline sympy/polys/polytools.py]: Only indent the docstring 4 spaces.

File: sympy/polys/polytools.py [hunk 3]
```diff
@@ -61,7 +61,40 @@
...
+
+        Examples
+        ========
+
+        >>> from sympy import Poly
+        >>> from sympy.abc import x, y
+
+        Create a univariate polynomial:
+
+        >>> Poly(x*(x**2 + x - 1)**2) #
```
Comment (smichr) [inline sympy/polys/polytools.py]: delete trailing space and pound sign

File: sympy/polys/polytools.py [hunk 4]
```diff
@@ -61,7 +61,40 @@
...
+        Poly(x**5 + 2*x**4 - x**3 - 2*x**2 + x, x, domain='ZZ')
+
+        Create a univariate polynomial with specified domain:
+
+        >>> Poly(x*(x**2 + x - 1)**2, domain = 'R')
```
Comment (smichr) [inline sympy/polys/polytools.py]: delete space around `=`

How about a more compact example, too?

```
>>> Poly((x+1)**2,domain=RR)
Poly(1.0*x**2 + 2.0*x + 1.0, x, domain='RR')
```

File: sympy/polys/polytools.py [hunk 5]
```diff
@@ -61,7 +61,40 @@
...
+        Poly(1.0*x**5 + 2.0*x**4 - 1.0*x**3 - 2.0*x**2 + 1.0*x, x, domain='RR')
+
+        Create a multivariate polynomial:
+
+        >> Poly(y*x**2 + x*y + 1)
```
Comment (smichr) [inline sympy/polys/polytools.py]: `>>` -> `>>>` or else this won't be tested

File: sympy/polys/polytools.py [hunk 6]
```diff
@@ -61,7 +61,40 @@
...
+        Poly(x**2*y + x*y + 1, x, y, domain='ZZ')
+
+        Create a univariate polynomial, where y is a constant. You can then evaluate this polynomial as a function of y:
```
Comment (smichr) [inline sympy/polys/polytools.py]: wrap long line to less than 72 chars

File: sympy/polys/polytools.py [hunk 7]
```diff
@@ -61,8 +61,44 @@
...
+
+    Examples
+    ========
+
+    >>> from sympy import Poly
+    >>> from sympy.abc import x, y
+
+    Create a univariate polynomial:
+
+    >>> Poly(x*(x**2+x-1)**2)
```
Comment (smichr) [inline sympy/polys/polytools.py]: Spaces around all unary operators (like `+` and `-`) here and below.

File: sympy/polys/polytools.py [hunk 8]
```diff
@@ -61,8 +61,44 @@
...
+    Poly(x**5 + 2*x**4 - x**3 - 2*x**2 + x, x, domain='ZZ')
+
+    Create a univariate polynomial with specified domain:
 
+    >>> Poly(x*(x**2+x-1)**2, domain='R')
```
Comment (smichr) [inline sympy/polys/polytools.py]: Please use a simple example here like

```
>>> Poly(x**2 + 2*x + sqrt(3), domain='R')
Poly(1.0*x**2 + 2.0*x + 1.73205080756888, x, domain='RR')
```

File: sympy/polys/polytools.py [hunk 9]
```diff
@@ -61,8 +61,44 @@
...
+    Poly(1.0*x**5 + 2.0*x**4 - 1.0*x**3 - 2.0*x**2 + 1.0*x, x, domain='RR')
+
+    Create a multivariate polynomial:
+
+    >>> Poly(y*x**2+x*y+1)
+    Poly(x**2*y + x*y + 1, x, y, domain='ZZ')
+
+    Create a univariate polynomial, where y is a constant:
+    Note: You can then evaluate this polynomial as a function of y.
```
Comment (smichr) [inline sympy/polys/polytools.py]: It's unclear what you mean by 'evaluate' here. Can you provide an example?

File: sympy/polys/polytools.py [hunk 10]
```diff
@@ -61,8 +61,44 @@
...
+
+    >>> Poly(y*x**2+x*y+1,x)
+    Poly(y*x**2 + y*x + 1, x, domain='ZZ[y]')
+
+    See Also
+    ========
+
```
Comment (smichr) [inline sympy/polys/polytools.py]: you can remove the blank line here. Most of the time you have to have it there, but with "See Also" you don't.

**Comment by smichr** [comment]
When you fix the trailing space error, could you squash all the commits into one? Then this is ready to go.  Also, I would recommend changing `specified domain` -> `specific domain`.

**Comment by smichr** [comment]
+1 Looks good.

**Comment by jtoka** [comment]
thanks for the reviews and for sticking with me on this @smichr

**Comment by smichr** [comment]
> thanks for the reviews and for sticking with me on this

No problem. It's going to be nice to have a docstring there. Thanks for getting this started!

File: sympy/polys/polytools.py [hunk 11]
```diff
@@ -61,7 +61,47 @@
 
 @public
 class Poly(Expr):
-    """Generic class for representing polynomial expressions. """
+    """
+    Generic class for representing and operating on polynomial expressions.
+    Subclasses Expr class.
```
Comment (thisch) [inline sympy/polys/polytools.py]: Do we really need to explicitly mention that this class subclasses from expr?

File: sympy/polys/polytools.py [hunk 12]
```diff
@@ -61,7 +61,47 @@
...
+    Poly(x**2*y + x*y + 1, x, y, domain='ZZ')
+
+    Create a univariate polynomial, where y is a constant:
+
+    >>> Poly(y*x**2 + x*y + 1,x)
+    Poly(y*x**2 + y*x + 1, x, domain='ZZ[y]')
+
+    You can evaluate the above polynomial as a function of y:
+
+    >>> Poly(y*x**2 + x*y + 1,x).eval(2)
```
Comment (thisch) [inline sympy/polys/polytools.py]: whitespace missing before `x)`

File: sympy/polys/polytools.py [hunk 13]
```diff
@@ -61,7 +61,47 @@
...
+    Poly(1.0*x**2 + 2.0*x + 1.73205080756888, x, domain='RR')
+
+    Create a multivariate polynomial:
+
+    >>> Poly(y*x**2 + x*y + 1)
+    Poly(x**2*y + x*y + 1, x, y, domain='ZZ')
+
+    Create a univariate polynomial, where y is a constant:
+
+    >>> Poly(y*x**2 + x*y + 1,x)
```
Comment (thisch) [inline sympy/polys/polytools.py]: whitespace before `x)`

**Pull request merged**